### PR TITLE
Add kaks_calculator2.0

### PR DESCRIPTION
**Pull request by gallardoalba**
Add package kaks_calculator
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/kaks_calculator/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  host:
+    - zlib
+
+test:
+  commands:
+    - KaKs_Calculator -h 2>&1 | grep "KaKs_Calculator" > /dev/null
+
+about:
+  home: 'http://sourceforge.net/projects/kakscalculator2/'
+  license: GPL-3.0-or-later
```
Comment (bgruening) [inline recipes/kaks_calculator/meta.yaml:28]: where did you found the License information? Is GPL or later correct?
We need to add the license_file as well

**Comment by bgruening** [comment]
> GY94.cpp: In member function 'int GY94::preProcess(const char*, const char*)':
GY94.cpp:229:18: error: 'strlen' was not declared in this scope
  229 |         com.ls = strlen(seq1);
      |                  ^~~~~~


It does not compile :(

**Comment by kullrich** [comment]
Hi, are you kidding me?
KaKs_Calculatro2 is already in bioconda:
https://bioconda.github.io/recipes/kakscalculator2/README.html?highlight=kaks_calculator2
Best regards
Kristian

**Comment by bgruening** [comment]
@kullrich thanks for pointing this out. I'm sure that @gallardoalba was not aware of this.

As a side note, no one wants to "kidding you". Maybe you can be more respectful to others in your communication.

**Comment by kullrich** [comment]
Hi @bgruening and @gallardoalba,
sorry for my wording.
The packacge is in the repository now since 3 years and I was thinking that it is aware to everyone and easily accessible.
If you might ask why GPL-3, I contacted the original authors of KaKs_Calculator2 and it was published under this licence originally.
In addition I needed to make small changes so that it compiles.
Best regards
Kristian

**Comment by bgruening** [comment]
No problem. I think the confusion comes from the different naming and that https://anaconda.org/search?q=kaks_calculator is not so smart in searching.

Thanks for contributing to this package and maintaining it. If you have contact with the original author you might want to point from the original sourceforge repo to your GitHub one.

**Comment by kullrich** [comment]
@bgruening Thanks, I was not aware that this is possible. I am going to contact them and ask if it is possible to add the link there.

**Comment by kullrich** [comment]
Hi @gallardoalba,
the whole KaKs_Calculator2 C++ part was also ported into the R-bioconductor Package:
https://bioconductor.org/packages/release/bioc/html/MSA2dist.html
And with this is also available via bioconda:
https://bioconda.github.io/recipes/bioconductor-msa2dist/README.html
This was done so that the R-community can use the dN/dS calculation part without any external dependencies.
Best regards
Kristian

**Comment by gallardoalba** [comment]
> Hi, are you kidding me? KaKs_Calculatro2 is already in bioconda: https://bioconda.github.io/recipes/kakscalculator2/README.html?highlight=kaks_calculator2 Best regards Kristian

Hi, sorry I just didn't find it because of the naming.
Regards

**Pull request closed without merging**